pkg/gatewayserver/io/udp/udp.go

```diff
@@ -383,16 +387,20 @@ func (s *srv) handleDown(ctx context.Context, state *state) error {
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
-		case <-state.io.Context().Done():
-			return state.io.Context().Err()
-		case down := <-state.io.Down():
+		case <-st.io.Context().Done():
+			return st.io.Context().Err()
+		case down := <-st.io.Down():
 			tx, err := encoding.FromDownlinkMessage(down)
 			if err != nil {
 				logger.WithError(err).Warn("Failed to marshal downlink message")
 				// TODO: Report to Network Server: https://github.com/TheThingsNetwork/lorawan-stack/issues/76
 				break
 			}
-			downlinkPath := state.lastDownlinkPath.Load().(downlinkPath)
+			downlinkPath := st.lastDownlinkPath.Load()
+			if downlinkPath == nil {
+				logger.Debug("Received downlink message without an active downlink path")
+				break
+			}
```

Comment:
How about?
```go
downlinkPath, ok := state.lastDownlinkPath.Load().(downlinkPath)
if !ok {
    logger.Debug("Received downlink message without an active downlink path")
}
```

Reply:
The value stored is no longer `downlinkPath`, but `*downlinkPath`, and there is no `interface{}` in between - the result of `Load` is of type `*downlinkPath`.